docs: contributors enhancements

kairoaraujo opened the pull request:
This commit simplifies the contributor's doc.
It adds the reference to the Secure Systems Lab Development Guidelines,
gives more evidence to the tox usage, shares information about the tests,
linting, and coverage, and creates a session about submitting
the contributions highlighted by the DCO.

Fixes #1709

Signed-off-by: Kairo de Araujo <kdearaujo@vmware.com>

- [ ] The code follows the [Code Style Guidelines](https://github.com/secure-systems-lab/code-style-guidelines#code-style-guidelines)
- [ ] Tests have been added for the bug fix or new feature
- [X] Docs have been added for the bug fix or new feature

Commit:
docs: contributors enhancements

This commit simplifies the contributor's doc.
It adds the reference to the Secure Systems Lab Development Guidelines,
gives more evidence to the tox usage, shares information about the tests,
linting, and coverage, and creates a session about submitting
the contributions highlighted by the DCO.

Fixes #1709

Signed-off-by: Kairo de Araujo <kdearaujo@vmware.com>

kairoaraujo commented:
I haven't changed the installation part because IMHO, it is straightforward for those willing to contribute to the code, documentation, etc.

lukpueh commented on docs/CONTRIBUTORS.rst:
```diff
@@ -3,90 +3,19 @@ Instructions for Contributors
 
+To facilitate development and installation of edited version of the code base,
+developers are encouraged to install
+`Virtualenv <https://virtualenv.pypa.io/en/latest/index.html>`_.
```
```suggestion
developers are encouraged to use
`venv <https://docs.python.org/3/library/venv.html>`_.
```

lukpueh commented on docs/CONTRIBUTORS.rst:
```diff
@@ -3,90 +3,19 @@ Instructions for Contributors
-
+Contributions are encouraged to read the `Secure Systems Lab Development Guidelines
+<https://github.com/secure-systems-lab/lab-guidelines/blob/master/dev-workflow.md>`_
+before `Submitting Contributions`_.
```
What about combining the first two paragraphs e.g. like so...

```
Contribute to python-tuf by submitting pull requests against the "develop"
branch of this repository. Detailed instructions are available in our
`development guidelines
<https://github.com/secure-systems-lab/lab-guidelines/blob/master/dev-workflow.md>`_.
All submitted code should follow our `style guidelines
<https://github.com/secure-systems-lab/code-style-guidelines/blob/master/python.md>`__
and must be unit tested [reference to test section below] .
```
?

lukpueh commented on docs/CONTRIBUTORS.rst:
```diff
@@ -105,96 +34,73 @@ To work on the TUF project, it's best to perform a development install.
-    $ cd tests
-    $ python3 aggregate_tests.py
+``tox`` will install ``tuf`` and dependencies, run the tests, linting, and
+coverage.
```
This used to be one sentence...

> While a supported Python version must already be available,``tox`` will install ``tuf`` and dependencies.

lukpueh commented on docs/CONTRIBUTORS.rst:
```diff
@@ -105,96 +34,73 @@ To work on the TUF project, it's best to perform a development install.
+you need debug/run outside ``tox``.
 
-All of the log levels and the corresponding options that could be used for testing are:
+tests
```
```suggestion
Unit Tests
```

kairoaraujo replied:
I fixed it, but IMO, Tests would be better once we don't have only unit tests running.

lukpueh commented on docs/CONTRIBUTORS.rst:
```diff
@@ -105,96 +34,73 @@ To work on the TUF project, it's best to perform a development install.
+See the the help (``-h``) for the usage.
+
+
+coverage
```
```suggestion
Coverage
```

lukpueh commented on docs/CONTRIBUTORS.rst:
```diff
@@ -105,96 +34,73 @@ To work on the TUF project, it's best to perform a development install.
+    $ cd tests
+    $ python3 tests/aggregate_tests.py
 
+See the the help (``-h``) for the usage.
```
Does `aggregate_tests` have a `-h` option?

kairoaraujo replied:
My fault, individual tests have `-h` option

lukpueh commented on docs/CONTRIBUTORS.rst:
```diff
@@ -105,96 +34,73 @@ To work on the TUF project, it's best to perform a development install.
+    $ cd tests/
     $ coverage run aggregate_tests.py && coverage report
 
+liniting
```
```suggestion
Auto-formatting
```

lukpueh reviewed (changes requested):
Great set of simplifications, @kairoaraujo! I have a few comments inline. Let me know what you think. 

@jku also suggested to mention how to run individual tests with additional options (see https://github.com/theupdateframework/python-tuf/issues/1709#issuecomment-995571198). Could you add a paragraph for that?

Also, since you are already restructuring most of the document. Would you mind converting it to markdown? We use it for many other docs and it feels just easier to edit.

kairoaraujo commented:
> Also, since you are already restructuring most of the document. Would you mind converting it to markdown? We use it for many other docs and it feels just easier to edit.

I did not convert to Markdown once we would need to add an extension to the sphinx and fix some formatting issues in other Markdown files that triggered a Warning. 
It would break our CI docs/readthedocs.org
 Should I file a new issue for this migration from RST to MD?

lukpueh commented:
> Should I file a new issue for this migration from RST to MD?

No, it's okay.

lukpueh commented on docs/CONTRIBUTORS.rst:
```diff
@@ -1,92 +1,22 @@
+<https://github.com/secure-systems-lab/lab-guidelines/blob/master/dev-workflow.md>`_.
+All submitted code should follow our `style guidelines
+<https://github.com/secure-systems-lab/code-style-guidelines/blob/master/python.md>`_
+and must be unit tested [reference to test section below] .
```
> [reference to test section below]
Hehe, I actually meant to make "unit tested" a link to the test section below. I didn't know the rst syntax form the top of my head. :D

kairoaraujo replied:
Oh, I'm sorry! Actually, it was a regression after I tried to migrate it to Markdown.

lukpueh commented on docs/CONTRIBUTORS.rst:
```diff
@@ -105,96 +35,77 @@ To work on the TUF project, it's best to perform a development install.
-`isort <https://pycqa.github.io/isort/docs/configuration/pre-commit/>`__].
+    $ tox
 
+Below, you will see more details about each steps managed by ``tox``, in case
```
```suggestion
Below, you will see more details about each step managed by ``tox``, in case
```

lukpueh commented on docs/CONTRIBUTORS.rst:
```diff
@@ -105,96 +35,77 @@ To work on the TUF project, it's best to perform a development install.
-    $ python3 test_updater_ng.py -vvvv TestUpdater.test_refresh_and_download # run test with DEBUG log level
 
+    $ cd tests/
+    $ python3 test_updater_ng.py -v
```
I think this should also be rendered in a code block like the above.

lukpueh commented on docs/CONTRIBUTORS.rst:
```diff
@@ -105,96 +35,77 @@ To work on the TUF project, it's best to perform a development install.
-test ``tuf`` against the tip of development of ``securesystemslib`` on GitHub,
-to e.g. prepare the former for a new release of the latter.
-::
+Contributions can be made by submitting GitHub pull requests.
```
I think this sentence is no longer necessary. We already say so above.

lukpueh commented on docs/CONTRIBUTORS.rst:
```diff
@@ -105,96 +35,77 @@ To work on the TUF project, it's best to perform a development install.
 
-    $ tox
 
+Submitting Contributions
+========================
```
This section is really about DCO, should we change the title?
```suggestion
Submitting Contributions
```
```suggestion
DCO
===
```

lukpueh commented on docs/CONTRIBUTORS.rst:
```diff
@@ -105,96 +35,77 @@ To work on the TUF project, it's best to perform a development install.
-<https://black.readthedocs.io/>`__ and `isort <https://pycqa.github.io/isort>`__.
-Auto-formatting can be done on the command line:
-::
+With `tox <https://testrun.org/tox/>`_, the test suite can be executed in a
+separate *venv*. While a supported Python version must already be available,
+``tox`` will install ``tuf`` and dependencies, run the tests, auto-formatting,
+and coverage.
```
I'm not a comma expert, but I think here's a superfluous one. Also, should we mention that tox runs tests for each Python version? E.g.:
```suggestion
With `tox <https://testrun.org/tox/>`_ the whole test suite can be executed in
a separate *virtual environment* for each supported Python version available on
the system. ``tuf`` and its dependencies are installed automatically for each tox
run.
```

kairoaraujo replied:
Thanks, I will look into the commas. Writing docs is a big challenge to me 🤪.

About the tox multiple versions, it doesn't run for all versions anymore in the default call. Not sure if we need to go deep on tox usage. The usage is in the link above.

lukpueh replied:
Oh right. Apologies for the noise then. Please disregard my suggestion.

Commit:
fixed links, RST formatting and text

Fixed some links, RST formatting and made some texts more clear
understanding.

Signed-off-by: Kairo de Araujo <kdearaujo@vmware.com>

jku commented on docs/CONTRIBUTORS.rst:
```diff
@@ -1,92 +1,22 @@
+To facilitate development and installation of edited version of the code base,
+developers are encouraged to use `venv <https://docs.python.org/3/library/venv.html>`_.
+
 1. First, `install non-Python dependencies <INSTALLATION.rst#non-python-dependencies>`_.
```
link is broken on RTD

kairoaraujo replied:
Thank you, @jku.
I used ``'sphinx.ext.autosectionlabel'``, not sure if we want to use it like that or add the absolute link.

lukpueh replied:
I lean towards using the absolute link. Like that it will not only render correctly on RTD, but also on GitHub, which [currently it doesn't](https://github.com/theupdateframework/python-tuf/blob/1ae438b30f015c57c4d4a2b89bb7d72d303cd6c1/docs/CONTRIBUTORS.rst#development-installation).

kairoaraujo replied:
I'm OK with that. I was trying to use the sphinx feature
I left it enabled in the case it is needed for label reference.

jku reviewed (approved):
Looks like a real improvement to me. I noticed one more broken link on RTD, but the changes in this PR seem good.

Commit:
Fixed broken link on RTD

Fix the RTD link using the absolute link.

Signed-off-by: Kairo de Araujo <kdearaujo@vmware.com>
